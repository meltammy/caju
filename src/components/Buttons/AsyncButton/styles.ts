import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const StyledSpinner = styled(Spinner)`
  width: 5px;
  height: 5px;
  border-width: 3px;
`;
