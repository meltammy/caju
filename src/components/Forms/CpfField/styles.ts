import ReactInputMask from "react-text-mask";
import styled from "styled-components";

export const Input = styled(ReactInputMask)`
  width: 100%;
  padding: 0 0.5rem;
  min-height: 2.25rem;
  background-color: ${({ theme }) => theme.colors.white};
  transition: all 0.2s ease-in-out 0s;

  vertical-align: middle;
  border: 1px solid rgba(36, 28, 21, 0.3);
  border-radius: 0.5rem;

  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: normal;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;
