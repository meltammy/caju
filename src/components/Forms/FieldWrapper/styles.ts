import styled from "styled-components";

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 0.75rem;
`;

export const Container = styled.div`
  width: 100%;
`;

export const Label = styled.label`
  width: 100%;
  font-weight: 600;
`;
