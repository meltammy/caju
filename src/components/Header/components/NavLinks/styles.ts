import styled from "styled-components";

export const NavItems = styled.a`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
