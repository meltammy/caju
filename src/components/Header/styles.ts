import styled from "styled-components";

export const Container = styled.header`
  background: ${({ theme }) => theme.colors.white};
  width: 100%;
  box-shadow: 4px 4px 16px 3px #0000001f;
  height: 4rem;
  position: fixed;
  top: 0;
  display: flex;
  align-items: center;
  padding: 0px 1.5rem;
  gap: 2rem;
  z-index: 1;
`;

export const Title = styled.h1`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
