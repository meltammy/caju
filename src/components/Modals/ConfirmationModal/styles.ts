import styled from "styled-components";

export const ModalContainer = styled.div`
  min-width: 32rem;
  max-width: 95vw;
  z-index: 999;
  padding: 1.5rem;
  background: white;
  border-radius: 0.5rem;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 1.5rem;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    flex-direction: column-reverse;
  }
`;
