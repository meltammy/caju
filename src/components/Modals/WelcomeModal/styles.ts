import styled, { keyframes } from "styled-components";
import Button from "~/components/Buttons/Button";

const coming = keyframes`
  0% {
    bottom: -73rem;
  }

  100% {
    bottom: 0;
  }
`;

export const ModalContainer = styled.div`
  position: relative;
  z-index: 999;

  width: 52rem;
  height: 90%;
  max-height: 30rem;
  background: white;
  border-radius: 0.5rem;
  padding: 1.5rem;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);

  b {
    color: ${({ theme }) => theme.colors.primary};
  }

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    position: absolute;
    width: 100%;
    min-width: 50vw;
    max-height: 36rem;
    margin-top: auto;

    display: grid;
    grid-template-rows: 1fr 10fr 1fr;

    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;

    animation: ${coming} 1s ease-in-out forwards;
  }
`;

export const StyledButton = styled(Button)`
  width: 60%;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    width: 100%;
  }
`;

export const Title = styled.h1`
  margin: 0;
  font-size: 3rem;
  color: ${({ theme }) => theme.colors.primary};

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    margin: 0% 0 2rem 0;
    font-size: 2rem;
    text-align: center;
  }
`;

export const DescriptionContainer = styled.div``;

export const Description = styled.p`
  font-size: 1rem;
  max-width: 60%;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    max-width: 100%;
    margin: 0 0 1rem 0;
    font-size: 0.875rem;
  }
`;

export const Illustration = styled.img`
  position: absolute;
  right: 5%;
  bottom: 5%;
  max-height: 60%;
  z-index: -1;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    max-height: 43%;
  }
`;
