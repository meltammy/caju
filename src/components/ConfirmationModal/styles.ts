import styled from "styled-components";
import { AsyncButton } from "../Buttons/AsyncButton";

export const ModalOverlay = styled.div<{ $isOpen: boolean }>`
  display: ${(props) => (props.$isOpen ? "flex" : "none")};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

export const ModalContainer = styled.div`
  background: white;
  border-radius: 8px;
  padding: 20px;
  max-width: 400px;
  width: 100%;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 20px;
`;

export const ButtonContainer = styled.div`
  display: grid;
  gap: 10px;
  grid-template-columns: 1fr 1fr;
`;

export const StyledAsyncButton = styled(AsyncButton)<{
  $isConfirmation?: boolean;
}>`
  padding: 10px 15px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  background: ${({ theme, $isConfirmation }) =>
    $isConfirmation ? theme.colors.primary : theme.colors.grey};
  color: white;
  height: 35px;

  &:hover {
    opacity: 0.8;
  }

  &:disabled {
    background: ${({ theme }) => theme.colors.grey};
    opacity: 1;
    cursor: progress;
  }
`;
