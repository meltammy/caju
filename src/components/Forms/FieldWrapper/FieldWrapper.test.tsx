import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { FieldWrapper } from ".";
import { ThemeProvider } from "styled-components";
import { theme } from "~/../styles/theme";

const helpId = "help-test";
const renderFieldWrapper = (
  props: Partial<React.ComponentProps<typeof FieldWrapper>> = {}
) => {
  const defaultProps = {
    id: "test-field",
    helpId,
    label: "Test Label",
    children: <input type="text" id="test-field" />,
    ...props,
  };

  return render(
    <ThemeProvider theme={theme}>
      <FieldWrapper {...defaultProps} />
    </ThemeProvider>
  );
};

describe("FieldWrapper component", () => {
  it("should render label and children", () => {
    renderFieldWrapper();

    expect(screen.getByLabelText(/test label/i)).toBeInTheDocument();

    const input = screen.getByLabelText(/test label/i);

    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute("id", "test-field");
  });

  it("should render error message when provided", () => {
    renderFieldWrapper({
      error: "This field is required",
    });

    const errorMessage = screen.getByTestId(helpId);

    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveTextContent("This field is required");
  });

  it("should does not render error message when not provided", () => {
    renderFieldWrapper();

    const errorMessage = screen.queryByTestId(helpId);
    expect(errorMessage).not.toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = renderFieldWrapper();
    expect(asFragment()).toMatchSnapshot();
  });
});
