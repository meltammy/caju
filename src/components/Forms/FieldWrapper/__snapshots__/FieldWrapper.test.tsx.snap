// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWrapper component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      aria-labelledby="test-field"
      for="test-field"
    >
      Test Label
    </label>
    <input
      id="test-field"
      type="text"
    />
  </div>
</DocumentFragment>
`;
